### PR TITLE
WIP towards MVP

### PR DESCRIPTION
**Pull request by cheneveld**
Next Up: 

1. Skills API request, local persistence, UI to edit, send with bounties request to filter
 - https://gitcoin.co/profile/cheneveld/keywords
2. Hook up Fabric / Mixpanel
3. Design/Implement Push Notification System
4. More better UI/UX

### CODE REVIEW

File: Gitcoin/DraggableViewBackground.m [hunk 1]
```diff
@@ -1,229 +0,0 @@
...
-
-- (id)initWithFrame:(CGRect)frame
-{
-    self = [super initWithFrame:frame];
-    if (self) {
-        [super layoutSubviews];
-        [self setupView];
-
-        NSError *error;
-        NSString *url_string = [NSString stringWithFormat: @"https://gitcoin.co/api/v0.1/bounties?&idx_status=submitted&order_by=-web3_created"];
```
Comment (owocki) [inline Gitcoin/DraggableViewBackground.m]: i think this should be `https://gitcoin.co/api/v0.1/bounties/?&idx_status=open&order_by=-web3_created`

i migrated all the 'submitted' statuses to be 'open'

File: README.md [hunk 2]
```diff
@@ -23,11 +24,14 @@ Gitcoin pushes Open Source Forward. Learn more at [https://gitcoin.co](https://g
...
+# Credits
 
-# credits
+Thanks to  [Koloda](https://github.com/Yalantis/Koloda).
 
-Thanks to Richard Kim for [RKSwipeCards](https://github.com/cwRichardKim/RKSwipeCards).
 
+# Setup
+1. Install cocoapods, and run pod install
+2. Create a SafeConfiguration.plist file and add the following keys as strings: gitHubOAuthToken, gitHubOAuthSecret
```
Comment (owocki) [inline README.md]: is it worth creating a `SafeConfiguration.plist.dist` file as a template?

**Comment by owocki** [comment]
👍

File: README.md (hunk 2 above)
Comment (cheneveld) [reply README.md]: Maybe? I am not familiar with the *.dist pattern...

the Xcode project will have the missing file -> https://screencast.com/t/fQ1R2bMbnC
and i suppose if this file gets larger we will need something

perhaps a script that generates the file or something?

Comment (owocki) [reply README.md]: the *.dist pattern is just a sample file with fake keys in it.

Comment (cheneveld) [reply README.md]: id say we just stick the sample in the readme... for now

File: Gitcoin/DraggableViewBackground.m (hunk 1 above)
Comment (cheneveld) [reply Gitcoin/DraggableViewBackground.m]: fixed in c7fc847

File: README.md (hunk 2 above)
Comment (cheneveld) [reply README.md]: fixed in c7fc847

**Comment by cheneveld** [comment]
@owocki the current keyword endpoint is ```https://gitcoin.co/profile/cheneveld/keywords```

This is not behind the "api".  

Could we create a route like https://gitcoin.co/api/v0.1/profile/cheneveld/keywords???

**Comment by cheneveld** [comment]
@owocki Could you have api errors respond with a 500 status code and a json formatted response?

https://gitcoin.co/api/v0.1/bounties/?idx_status=open&order_by=-web3_created&pk__gt=

**Comment by owocki** [comment]
> @owocki Could you have api errors respond with a 500 status code and a json formatted response?

> Could we create a route like https://gitcoin.co/api/v0.1/profile/cheneveld/keywords???

just pushed up a fix for both of these

**Comment by owocki** [comment]
Looks like I'm missing something called 'Mapper'.  Does it need to be added to the podfile? http://bits.owocki.com/051m3M0G3i0N/Image%202017-11-09%20at%2011.12.15%20AM.png

**Comment by cheneveld** [comment]
No you probably opened the "Gitcoin.xcodeproj" file, need to open and use the workspace file "Gitcoin.xcworkspace"

**Comment by owocki** [comment]
> No you probably opened the "Gitcoin.xcodeproj" file, need to open and use the workspace file "Gitcoin.xcworkspace"


hmmm i seem to have a bunch more build issues when i do this => http://bits.owocki.com/3k3W0I2K2m0z/Screen%20Shot%202017-11-09%20at%201.13.32%20PM.png

it's not super urgent for me to get this working; maybe next time we get together we can pair on getting it working

**Comment by cheneveld** [comment]
k, what version of xcode are you on?

**Comment by owocki** [comment]
Version 9.1 (9B55)

**Comment by cheneveld** [comment]
Pull this latest and rerun pod install

**Comment by cheneveld** [comment]
I think we will need to convert from cocoapods to either Swift Package Manager or Carthage for our dependency management in the future.

I just got some weird behavior with pods.

after I commented out the pod dependencies in the [Podfile](https://github.com/cheneveld/ios/blob/master/Podfile#L10-L33), ran pod install, uncommented em, then pod install again it started working again.

@owocki perhaps do that if you still have problems.

I think long term is to move away from Cocoapods

**Comment by owocki** [comment]
it built!  i'm in, thanks!

**Comment by cheneveld** [comment]
@owocki I'm ready to pass the user skills/keywords to the [bounty api endpoint via raw_data](https://github.com/gitcoinco/ios/pull/3/commits/c36dbfb863acd9e34f1ebcce42cb3f92dd54cf8e#diff-416c854fa58ce60ef5ba98ea312f2bf6R100) but seems it only accepts a single string.  How can I pass multiple values?

**Comment by owocki** [comment]
> @owocki I'm ready to pass the user skills/keywords to the bounty api endpoint via raw_data but seems it only accepts a single string. How can I pass multiple values?

Just pushed support for `comma,delimited,keyword,values`.  Give it a try now!

**Comment by owocki** [comment]
heres a sample request from my local:

```
http://localhost:8080/api/v0.1/bounties/?&raw_data=owocki,piper&idx_status=open&network=mainnet&coinbase=0x4331b095bc38dc3bce0a269682b5ebaefa252929&order_by=-web3_created
```

note the `owocki,piper`

**Comment by cheneveld** [comment]
@owocki can we have another bounty description field that is plain text, removed html, for displaying simply on the bounty cards?

**Comment by owocki** [comment]
> @owocki can we have another bounty description field that is plain text, removed html, for displaying simply on the bounty cards?

sure, itll be called `issue_description_text`

**Comment by owocki** [comment]
deploying `issue_description_text` now

**Comment by cheneveld** [comment]
@owocki could you clean up the issue_description_text? I think just removing newlines at begining of text and removing when we have 3 or more turn to 2 -> https://screencast.com/t/vbUqdeOo5

**Comment by owocki** [comment]
> @owocki could you clean up the issue_description_text? I think just removing newlines at begining of text and removing when we have 3 or more turn to 2 -> https://screencast.com/t/vbUqdeOo5

sure thing, https://github.com/gitcoinco/web/commit/54e728adf341c6e83b1a8bff2b04115001ed9002 takes care of this

**Comment by cheneveld** [comment]
Getting down to it.  

Next up is more UI and COPY tweaks. 
Then review TODOs through out app.  
Refine tracking.

**Comment by cheneveld** [comment]
@owocki 

Thinking about this project being capable of allowing any developer to get going in xcode right away... as it stands now it is pointing to production by default.  Is this ok? or should we get hooked into staging? 

Perhaps we hook in with the creds in the project?

Thoughts?

Perhaps flags on the api calls to say that its a debug build and to not send the intro emails?

**Comment by owocki** [comment]
i think production is fine.

**Comment by owocki** [comment]
> Perhaps flags on the api calls to say that its a debug build and to not send the intro emails?

sure. how about `&debug_build=1`

**Comment by cheneveld** [comment]
or how bought pretend=1 ?

**Comment by owocki** [comment]
works 4 me

**Comment by owocki** [comment]
i fixed the match emails tongiht http://bits.owocki.com/0u0D0z3v3F3B/Screen%20Shot%202017-12-01%20at%2011.22.01%20PM.png

**Comment by cheneveld** [comment]
@owocki per Alpha feedback ```you shouldnt force users to sign in to be able to accumulate some bounties```...

How bought we let the user swipe left (-) while signed out.  But if a swipe right occurs when the user is logged out we display an alert with some intel on what that action is actually doing.  Like "Please sign in to gihub so we can connect you with the repo owner of this bounty".

Thoughts?

**Comment by owocki** [comment]
> How bought we let the user swipe left (-) while signed out. But if a swipe right occurs when the user is logged out we display an alert with some intel on what that action is actually doing. Like "Please sign in to gihub so we can connect you with the repo owner of this bounty".

I could go either way on this.   Slightly in favor of this.

> Also, Everytime you login or logout the lastBountyId is reset (wonder if we can make this obvious some how?). So perhaps we can flash that alert recommend logging in and then allowing the right swipe with after the alert is shown once.

Maybe we could put some copy below the 'sign out' button that says as much.

**Comment by cheneveld** [comment]
@owocki Could you provide some copy -> https://screencast.com/t/oMZpskyYS6b

**Comment by cheneveld** [comment]
per 

> it'd be really cool if we could enumerate the users 1 by 1 and see their attributes (prog languages, num new bounties since last push)… to send a super custom push notification

Now sending pushwoosh user info and events. so we can play with events, auto pushes and such.

**Comment by owocki** [comment]
> @owocki Could you provide some copy -> https://screencast.com/t/oMZpskyYS6b

h1- Sign In with Github
p - to get match email intros to this repo owner.

> Now sending pushwoosh user info and events. so we can play with events, auto pushes and such.

neat, what user info / events are you sending?

**Comment by cheneveld** [comment]
Check out this commit -> https://github.com/gitcoinco/ios/pull/3/commits/411dabda5848d4e0edec1a1195dcc7dcb3ef8723

File: Gitcoin/Util/TrackingManager.swift [hunk 3]
```diff
@@ -31,99 +32,113 @@ class TrackingManager {
...
+                    for userField in  userDictionary {
+                        
+                        if let userFieldValue = userField.value as? String {
+                            customAttributes["user-\(userField.key)"] = userFieldValue
+                        }
+                    }
+                }
+                
+                Answers.logCustomEvent(withName: "didEditKeywords", customAttributes: customAttributes)
+                Mixpanel.mainInstance().track(event: "didEditKeywords", properties: customAttributes)
```
Comment (cheneveld) [inline Gitcoin/Util/TrackingManager.swift]: @owocki here is us sending all the interactions with keywords to our 3 services

**Comment by cheneveld** [comment]
@owocki noticed the web app stores keywords?? or looks like it -> https://screencast.com/t/7eXDKOThMeu.  

Should we think about having the keywords insync with ios?

**Comment by owocki** [comment]
> Should we think about having the keywords insync with ios?

for 99% of UX, this seems 👍 to me.  but i worry that having them in sync with each other would introduce an attack vector wherein a user on their iphone could enter an erroneous email address and update anyone's keywords.  i suppose we'd have to have an email address confirmation to close this attack vector.

**Comment by owocki** [comment]
wait, on second thought... if we use github oauth, by definition, that will always just have a confirmed email address

**Pull request merged**